### DISCUSSION TITLE
Exception when configuring MsSqlTestcontainer

### DISCUSSION THREAD

**Discussion by EnisMulic** [testcontainers/testcontainers-dotnet#439] [Q&A] [answered]
When trying to set `Username` and/or `Database` properties in `MsSqlTestcontainerConfiguration` I get the exception
`Object reference not set to an instance of an object.` is this an expected behavior?

```csharp
var config = new MsSqlTestcontainerConfiguration()
{
    Port = 1433,
    Password = "Pa$$w0rd!",
    Username = "TestUser",
    Database = "TestDb"
};


var container = new TestcontainersBuilder<MsSqlTestcontainer>()
    .WithDatabase(config)
    .WithExposedPort(1433)
    .WithPortBinding(1433, 1433)
    .Build();
```

**Comment by HofmeisterAn** [comment, marked as answer]
That's on purpose. As far as I know, you can not set the username and database through the MSSQL container start:
https://github.com/HofmeisterAn/dotnet-testcontainers/blob/859826e98e619000ffe5029f088ed9db86946d6f/src/DotNet.Testcontainers/Configurations/Modules/Databases/MsSqlTestcontainerConfiguration.cs#L44

You need to seed your database afterwards.
